### PR TITLE
Added read/unread state to the backend

### PR DESCRIPTION
**Pull request by Mo-sissoko**
### ℹ️ Issue

Closes # https://trello.com/c/d19gG2da/18-implement-a-read-unread-state-backend

### 📝 Description

Briefly list the changes made to the code:

- Added new endpoint to mark form as “read”
- Added new db utils function to mark form as "read"
- Added handler to handle marking form as "read"

### ✔️ Verification

What steps did you take to verify your changes work? These should be clear enough for someone to be able to clone the branch and follow the steps themselves. 

- Create new stack and check AWS Lambda endpoint for my new endpoint
Provide screenshots of any new components, styling changes, or pages. 

### 🏕️ (Optional) Future Work / Notes

Did you notice anything ugly during the course of this ticket? Any bugs, design challenges, or unexpected behavior? Write it down so we can clean it up in a future ticket!

### CODE REVIEW

File: apps/backend/src/handlers/markFormAsRead.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import { APIGatewayEvent } from 'aws-lambda';
+import { createTableIfNotExists } from '../db/createTable.js';
+import { markFormAsRead } from '../db/utils.js';
+import { createResponse } from '../utils/createResponse.js';
+import { validateMethodType } from '../utils/validateMethodType.js';
+import { validateRequestBody } from '../utils/validateRequestBody.js';
+import { formSchema } from '../schema/schema.js';
+/**
+ * An HTTP post method to add one form to the QLDB table.
```
Comment (nate-kirschner) [inline apps/backend/src/handlers/markFormAsRead.ts]: Comment needs updating

File: apps/backend/src/handlers/markFormAsRead.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    return response;
+  }
+
+  // All log statements are written to CloudWatch
+  console.info('received:', event);
+
+  const id = event.pathParameters!.id!;
+  const JSONBody = JSON.parse(event.body!);
+
+  const isValidBody = validateRequestBody(JSONBody, formSchema);
```
Comment (nate-kirschner) [inline apps/backend/src/handlers/markFormAsRead.ts]: I see that this is also a problem in the `patchFormHandler` function, but this is named incorrectly right? It should be isInvalidBody?

Comment (nate-kirschner) [inline apps/backend/src/handlers/markFormAsRead.ts]: Are we passing the entire function in the body of this request? Seems like only the id and a 'read' field are needed.

File: apps/backend/src/handlers/patchForm.ts [hunk 3]
```diff
@@ -32,6 +32,11 @@ export const patchFormHandler = async (event: APIGatewayEvent) => {
   try {
     await createTableIfNotExists();
     await updateDocumentAdminNotes(id, JSONBody);
+      // Check if the "read" property is included in the request body
```
Comment (nate-kirschner) [inline apps/backend/src/handlers/patchForm.ts]: Why do we need to do this here if we already have the other endpoing?

Also, this patchForm function seems to be pretty poorly named, but it seems like it's just for updating the admin notes, and updateDocumentAdminNotes expects the JSONBody to be an object that matches the adminNotesSchema, so read will never be in there.

Comment (Mo-sissoko) [reply apps/backend/src/handlers/patchForm.ts]: Looks like I forgot to remove it from there. I removed it and named this function better in my most recent commit.

File: apps/backend/src/handlers/markFormAsRead.ts (hunk 2 above)
Comment (Mo-sissoko) [reply apps/backend/src/handlers/markFormAsRead.ts]: Good point. I updated it to use the readSchema I created before.

Comment (Mo-sissoko) [reply apps/backend/src/handlers/markFormAsRead.ts]: Changed this to be "isInvalidBody" in my most recent commit.

File: apps/backend/src/handlers/markFormAsRead.ts (hunk 1 above)
Comment (Mo-sissoko) [reply apps/backend/src/handlers/markFormAsRead.ts]: Updated comment in recent commit.

File: apps/backend/src/schema/schema.ts [hunk 4]
```diff
@@ -64,4 +68,5 @@ export const formSchema = z.object({
   guardianForm: guardianFormSchema,
   medicalForm: medicalFormSchema,
   adminNotes: adminNotesSchema,
+  read: readSchema.optional(),
```
Comment (nate-kirschner) [inline apps/backend/src/schema/schema.ts]: Do we need the readSchema to actually be in the formSchema. Or can it just be an optional boolean, and then readSchema would only be used in the new handler. Seems redundant to have forms pass in the id in two locations, and also less intuitive for the frontend to implement

Comment (nate-kirschner) [reply apps/backend/src/schema/schema.ts]: Actually, even the readSchema shouldn't need an id since it gets passed in as a parameter in the request right?

Comment (Mo-sissoko) [reply apps/backend/src/schema/schema.ts]: Yeah, you're right it does get passed in. I'll change it.

Comment (Mo-sissoko) [reply apps/backend/src/schema/schema.ts]: Updated.

**Review by nate-kirschner — APPROVED** [review]

**Pull request merged**